### PR TITLE
update bayes ab

### PR DESCRIPTION
**Pull request by double-y**
### Description
update bayes ab

* indicate prior distribution with mean and sd
* change lift output to histogram
* change input from success count to success rate

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: R/do_bayes_ab.R [hunk 1]
```diff
@@ -1,47 +1,85 @@
...
 #' * model - Returns a data frame with bayesTest model.
 #' * summary - Output summary of the result of the test.
 #' * prior - Output coordinates of prior density chart.
 #' * posteriors - Output coordinates of posterior density chart of the success rate.
-#' * lift - Output probability density chart of lift, which is the ratio of performance improvement of A over B. The formula is (A - B) / B.
+#' * improvement - Output coordinate of histogram of lift, which is the ratio of performance improvement of A over B. The formula is (A - B) / B.
 #' @param seed Random seed for bayes test to estimate probability density.
 #' @export
-do_bayes_ab <- function(df, group, total_count, success_count, alpha = 1, beta = 1, type = "model", seed = 0, ...){
+do_bayes_ab <- function(df, a_b_identifier, total_count, success_rate, prior_mean = 0.5, prior_sd = 0.288675, type = "model", seed = 0, ...){
```
Comment (hideaki) [inline R/do_bayes_ab.R]: What is the reason for the prior_sd value? Please explain in the comment.

File: R/do_bayes_ab.R [hunk 2]
```diff
@@ -1,47 +1,85 @@
...
+
+  # validate alpha and beta
+  # when they are invalid, sd is too large
+  if (!(!is.na(alpha) && !is.na(beta) && alpha > 0 && beta > 0)){
+    stop("sd for prior is too large")
+  }
+
+  # when type is prior, no need to evaluate other parameters
+  if (type != "prior") {
+    a_b_identifier_col <- dplyr::select_var(names(df), !! rlang::enquo(a_b_identifier))
```
Comment (hideaki) [inline R/do_bayes_ab.R]: Can you explain what is done here?
What are !! and enquo for?

File: R/do_bayes_ab.R [hunk 3]
```diff
@@ -98,7 +141,7 @@ calc_beta_prior <- function(df, rate, ...){
 #' @param percentLift Lift threshold to calculate the probability of success
 #' @param credInt Ratio for credible interval
 #' @param type Type of output
-#' This can be "summary", "prior", "posteriors" and "lift"
+#' This can be "summary", "prior", "posteriors" and "improvement"
 #' @export
 tidy.bayesTest <- function(x, percentLift = 0, credInt = 0.9, type = "lift", ...) {
```
Comment (hideaki) [inline R/do_bayes_ab.R]: Shouldn't the default be renamed to "improvement" from "lift" too?

File: R/do_bayes_ab.R (hunk 1 above)
Comment (double-y) [inline R/do_bayes_ab.R]: Ok, will add a comment

File: R/do_bayes_ab.R (hunk 2 above)
Comment (double-y) [inline R/do_bayes_ab.R]: Ok, will add a comment

File: R/do_bayes_ab.R (hunk 3 above)
Comment (double-y) [inline R/do_bayes_ab.R]: right, will change it

**Pull request merged**